components/sup/src/main.rs

```diff
@@ -167,7 +167,9 @@ fn sub_run_rsr_imlw_mlw_gsw(m: &ArgMatches,
     set_supervisor_logging_options(m);
 
     let cfg = mgrcfg_from_sup_run_matches(m, feature_flags)?;
-    let manager = Manager::load_imlw(cfg, launcher)?;
+    let sys_ip = util::determine_sys_ip_address(m.value_of("SYS_IP_ADDRESS"))?;
```

Comment:
OPTIONAL: Assuming we do CLAP-level validation first, adding something like
```rust
fn ipaddr_from_matches(matches: &ArgMatches<'_>, arg_name: &str) -> Option<std::net::IpAddr> {
    matches.value_of(arg_name)
           .and_then(|s| std::net::IpAddr::from_str(s).ok())
}
```
would let us pass an `Option<IpAddr>` into determine_sys_ip_address, obviating the need for that function to handle a possibly invalid string.

Reply:
Addressed in [be3aa8](https://github.com/habitat-sh/habitat/pull/6960/commits/be3aa875b5d92f8bfbc6d34142579b2ba2e55d0e).